web/src/app/monitors/monitor-create-edit/monitor-create-edit.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { v4 as uuid } from 'uuid';
+
+// Third party modules
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs';
+
+// Dashboard hub models and services
+import { MonitorService, ProjectService } from '../../core/services/index.service';
+import { MonitorModel, ProjectModel } from '../../shared/models/index.model';
+
+@Component({
+  selector: 'dashboard-monitor-create-edit',
+  templateUrl: './monitor-create-edit.component.html',
+  styleUrls: ['./monitor-create-edit.component.scss'],
+})
+export class MonitorCreateEditComponent implements OnInit {
+
+  private monitorUid: string;
+  private monitorsList: MonitorModel[] = [];
+  private projectSubscription: Subscription;
+  private projectUid: string;
+  public isEdit: Boolean = false;
+  public monitorForm: FormGroup;
+
+  constructor(
+    private form: FormBuilder,
+    private monitorService: MonitorService,
+    private projectService: ProjectService,
+    private route: ActivatedRoute
+  ) { }
+
+  /**
+   * Lifecycle init method
+   */
+  ngOnInit(): void {
+    this.projectUid = this.route.snapshot.paramMap.get('projectUid');
+    this.intializeMonitorForm();
+    this.projectSubscription = this.projectService
+      .findOneById(this.projectUid)
```

Comment:
Now the resolver is on the route, it has the information already, so we don't need to fetch it again

Reply:
okay done